align_data/stampy/stampy.py

```
@@ -12,56 +17,41 @@
+        logger.info('Fetching table: %s', self.coda_doc_id)
         table = doc.get_table(self.on_site_table)
-        entries = table.to_dict() # a list of dicts
-        #
-        # keys for each element of the list of dicts: 
-        # {'Edit Answer', 'Status', 'aisafety.info Link', 'External Source', 'Tags', 'Related IDs', 'Alternate Phrasings', 'Rich Text', 'Doc Last Edited', 'Question', 'Link', 'UI ID', 'Related Answers'}
-        for ii, entry in enumerate(tqdm(entries)):
-            if self._entry_done(entry.get('Question','')):
-                # logger.info(f"Already done {entry}")
-                continue
-
-            question = entry.get('Question', '')
-            answer = entry.get('Rich Text', '')  # Assuming 'Answer' is present in each dict
-            url = entry.get('Link', '')
-            date_published = entry.get('Doc Last Edited', '')
-
-            # Creating the text field
-            #text = f"Question: {question}\n\nAnswer: {answer}" note: we don't do this because the title contains the question already
-
-            logger.info(f"Processing {ii}")
-
-            new_entry = DataEntry({
-                "source": self.name,
-                "source_filetype": "text",
-                "url": url,
-                "title": question,
-                "authors": "n/a",
-                "date_published": date_published,
-                "text": answer,
-                "question": question,
-                "answer": answer,
-                #"entry": entry  # This is the entire dictionary entry
-            })
-
-            logger.info(f"Processing {new_entry}")
-            new_entry.add_id()
-
-            yield new_entry
+        return table.to_dict() # a list of dicts
+
+    def get_item_key(self, entry):
+        return entry.get('Question', '')
+
+    def process_entry(self, entry):
+        question = entry.get('Question', '')
+        answer = entry.get('Rich Text', '')  # Assuming 'Answer' is present in each dict
```

I'd like some error handling for this, or at least some logging, but on the other hand it seems that errors here are a bigger problem

---

I'm thinking of modifying the question line to question = entry['question'] so an error appears if there's no question for the entry (which should not happen for the on site answers table). 
Although since this applies to other keys (answer, url, date_published), I can make this change to the other keys as well. Something's definitely wrong if there is no "Doc Last Edited" key for example.

---

remove data/stampy.jsonl and rerun it. The code will ignore entries that have already been processed :)

---

Is there a reason why we wouldn't want data/stampy.jsonl and data/stampy.txt to be deleted right before entering the new data? like when we run our program which wants to create these files, they should replace whatever was there beforehand, instead of silently doing nothing

---

but thank you, I see that it works

---

depends. In this case maybe so, but some data sources take a long time (e.g. lesswrong posts), so it's nice for it to not reprocess stuff if not needed. It might be worth discussing whether this should be controlled with a flag or something - worth raising this during the meeting